docs/docs/getting-started/developing-on-hydra.md

```diff
@@ -0,0 +1,89 @@
+
+However it is possible to spin up a cardano-node, and a cluster of hydra-nodes programmatically and run tests directly on that.
+The module hydra-cluster provides functions to spin up the set of required nodes and to run assertions on their output.
+
+A couple of steps are required to set up the initial environment:
+
+- We need to spin up at least one Cardano BFT node. This can be done with `withBFTNode` which takes the cardano node configuration and a callback that has access to an instance of the running node. The node parameters can be generated by a helper also exported by hydra-cluster: `newNodeConfig`.
+
+- In the callback, we can then spin up a cluster of hydra-nodes using `withHydraCluster`. This will require the cardano-node socket, which can be obtained from the running node instance, and a set of cardano and hydra keys which can also be easily generated through helpers exposed by `hydra-cluster`.
+
+Let's walk through a slightly simplified version of one of our tests, with comments explaining the setup we just described.
+This test will spin up a single cardano node, and a pair of hydra nodes (Alice and Bob).
+
+```haskell
+-- Tracer is an instance of https://hackage.haskell.org/package/contra-tracer and is used by several
+-- hydra components to deal with logs in a structured way
+it "spins up two hydra nodes" $ \tracer ->
+  -- Set a timeout for the whole test to fail
+  failAfter 60 $
+    -- We create a temporary directory to contain all the files required to spin up a node
+    withTempDir "cardano-node-tmp-dir" $ \cTmpDir -> do
+      config <- newNodeConfig cTmpDir
+      -- This starts the BFTNode as described above
+      withBFTNode (contramap FromCardanoNode tracer) config $ \(RunningNode _ nodeSocket) -> do
+        -- We create a temporary directory to contain all the files required to spin up the hydra nodes
+        withTempDir "hydra-nodes-tmp-dir" $ \hTmpDir -> do
+          -- We generate pairs of cardano keys for alice and bob, this pair is a tuple of verification and signing key
+          aliceKeys <- generate genKeyPair
+          bobKeys <- generate genKeyPair
+
+          -- Here we generate signing keys for hydra, note these are different from the cardano skeys generated above
+          let aliceSk = generateSigningKey "alice"
+              bobSk = generateSigningKey "bob"
+
+          let cardanoKeys = [aliceKeys, bobKeys]
+              hydraKeys = [aliceSk, bobSk]
+
+          -- Hydra nodes take an id as one of their arguments, it is used to identify them with the head protocol
+          let firstNodeId = 0
+
+          -- This will spin up `n` hydra nodes, where `n` is the length of the cardanoKeys list
+          -- (which needs to match the length of hydraKeys)
+          withHydraCluster tracer hTmpDir nodeSocket firstNodeId cardanoKeys hydraKeys $ \nodes -> do
+            ...
```

Comment:
If you do a `pure ()` here, we could even think of making this whole thing doctestable / compilable?

I think this is missing distributing funds with the `seedFromFaucet` stuff and maybe we also want to include some actual `send "Init" ["contestationPeriod" .= 10]` etc.?

Your text below could become comments in between.